homeassistant/components/blink/services.py

```diff
@@ -3,171 +3,76 @@
-            )
-
-        for coordinator in collect_coordinators(call.data[ATTR_DEVICE_ID]):
-            all_cameras = coordinator.api.cameras
-            if camera_name in all_cameras:
-                try:
-                    await all_cameras[camera_name].video_to_file(video_path)
-                except OSError as err:
-                    raise ServiceValidationError(
-                        str(err),
-                        translation_domain=DOMAIN,
-                        translation_key="cant_write",
-                    ) from err
-
-    async def async_handle_save_recent_clips_service(call: ServiceCall) -> None:
-        """Save multiple recent clips to output directory."""
-        camera_name = call.data[CONF_NAME]
-        clips_dir = call.data[CONF_FILE_PATH]
-        if not hass.config.is_allowed_path(clips_dir):
-            raise ServiceValidationError(
-                translation_domain=DOMAIN,
-                translation_key="no_path",
-                translation_placeholders={"target": clips_dir},
-            )
-
-        for coordinator in collect_coordinators(call.data[ATTR_DEVICE_ID]):
-            all_cameras = coordinator.api.cameras
-            if camera_name in all_cameras:
-                try:
-                    await all_cameras[camera_name].save_recent_clips(
-                        output_dir=clips_dir
-                    )
-                except OSError as err:
-                    raise ServiceValidationError(
-                        str(err),
-                        translation_domain=DOMAIN,
-                        translation_key="cant_write",
-                    ) from err
-
-    async def send_pin(call: ServiceCall):
-        """Call blink to send new pin."""
-        for coordinator in collect_coordinators(call.data[ATTR_DEVICE_ID]):
             await coordinator.api.auth.send_auth_key(
                 coordinator.api,
                 call.data[CONF_PIN],
             )
 
     async def blink_refresh(call: ServiceCall):
         """Call blink to refresh info."""
-        for coordinator in collect_coordinators(call.data[ATTR_DEVICE_ID]):
+        for entry_id in call.data[ATTR_CONFIG_ENTRY_ID]:
+            try:
+                coordinator = hass.data[DOMAIN][entry_id]
+            except KeyError as ex:
+                raise ServiceValidationError(
+                    translation_domain=DOMAIN,
+                    translation_key="integration_not_found",
+                    translation_placeholders={"target": DOMAIN},
+                ) from ex
+            if coordinator.config_entry.state != ConfigEntryState.LOADED:
+                raise HomeAssistantError(
+                    translation_domain=DOMAIN,
+                    translation_key="not_loaded",
+                    translation_placeholders={"target": coordinator.config_entry.title},
+                )
             await coordinator.api.refresh(force_cache=True)
```

Comment:
We should create a repair issue when this service is used and ask users to change to the `homeassistant.update_entity` service.

Search for issues that set `is_fixable=True` and marks a deprecated service, for examples.

Reply:
Previous comment said to do this in a separate PR.   I'll put it in this one.

Reply:
It's ok to do separately also.

Reply:
Might as well get it in now.